src/ApplicationSampler.cpp

```diff
@@ -182,8 +182,7 @@ namespace geopm
     void ApplicationSamplerImp::update(const geopm_time_s &curr_time)
     {
         if (!m_status || !m_sampler) {
-            throw Exception("ApplicationSamplerImp::" + std::string(__func__) + "(): cannot read process info before connect().",
-                            GEOPM_ERROR_RUNTIME, __FILE__, __LINE__);
+            return;
```

Comment:
It's too bad there isn't a way to tell the ApplicationSampler to get out of the way without removing these exceptions.  Turning them into no-ops seems like it might come back to bite us later if we're chasing app connect / data flow issues.

Maybe this is fine since app connect is pretty well ironed out at this point, and we don't expect any issues.

Reply:
Yeah, I don't see a better solution.